google/cloud/spanner_v1/snapshot.py

```diff
@@ -447,34 +452,38 @@ def execute_sql(
         if self._transaction_id is None:
             # lock is added to handle the inline begin for first rpc
             with self._lock:
-                iterator = _restart_on_unavailable(
-                    restart,
-                    request,
-                    "CloudSpanner.ReadWriteTransaction",
-                    self._session,
-                    trace_attributes,
-                    transaction=self,
+                return self._get_streamed_result_set(
+                    restart, request, trace_attributes, False
                 )
-                self._read_request_count += 1
-                self._execute_sql_count += 1
-
-                if self._multi_use:
-                    return StreamedResultSet(iterator, source=self)
-                else:
-                    return StreamedResultSet(iterator)
         else:
-            iterator = _restart_on_unavailable(
-                restart,
-                request,
-                "CloudSpanner.ReadWriteTransaction",
-                self._session,
-                trace_attributes,
-                transaction=self,
+            return self._get_streamed_result_set(
+                restart, request, trace_attributes, True
             )
 
+    def _get_streamed_result_set(
+        self, restart, request, trace_attributes, transaction_id_set
+    ):
+        iterator = _restart_on_unavailable(
+            restart,
+            request,
+            "CloudSpanner.ReadWriteTransaction",
+            self._session,
+            trace_attributes,
+            transaction=self,
+        )
         self._read_request_count += 1
         self._execute_sql_count += 1
 
+        if self._read_only and not transaction_id_set:
+            peek = next(iterator)
```

Comment:
Will this fail if the query returns no results?  (IIRC `next()` with no second/default argument throws a `StopIteration` exception or something if there's nothing left to return?)  Should it?

Reply:
There would be one result (PartialResultSet object) where `metadata` field would be present and `values` would be an empty list